packages/ra-core/src/sideEffect/accumulate.spec.js

```diff
@@ -0,0 +1,142 @@
+import { delay } from 'redux-saga';
+import { call, cancel, fork, put } from 'redux-saga/effects';
+import { createMockTask } from 'redux-saga/utils';
+
+import { accumulateFactory, finalizeFactory } from './accumulate';
+import { crudGetMany } from '../actions';
+
+describe('accumulate saga', () => {
+    describe('backward compatibility', () => {
+        it('proceeds with the accumulated action when no more actions are dispatched', () => {
+            const finalize = finalizeFactory({}, {});
+            const tasks = {};
+            const accumulations = {};
+
+            const saga = accumulateFactory(tasks, accumulations, finalize)({
+                payload: { resource: 'posts', ids: [1, 2] },
+                meta: { accumulate: crudGetMany },
+            });
+
+            expect(saga.next().value).toEqual(
+                fork(finalize, 'posts', crudGetMany)
+            );
+
+            expect(accumulations).toEqual({
```

Comment:
could you also test the `tasks` value?

Reply:
Nope, because it will be `{ posts: undefined }` as `fork` returns that in tests. That's actually why I had to use `createMockTask` to test task cancellation

Reply:
gotcha